query.go

```diff
@@ -438,6 +438,12 @@ func (c *Conn) Query(sql string, args ...interface{}) (*Rows, error) {
 		}
 	}
 
+	// Set query execution deadline
+	if c.config.QueryExecTimeout > 0 {
+		c.conn.SetDeadline(time.Now().Add(c.config.QueryExecTimeout))
+		defer c.conn.SetDeadline(time.Time{}) // reset deadline
```

Comment:
You're right, the connection is not going to be Release()'d in this method, I must have been looking somewhere else...

I guess it's perfectly safe for `c.conn` to be modified here.